File: src/ch/eif/intelliprolog/module/PrologModuleType.java
package ch.eif.intelliprolog.module;

import ch.eif.intelliprolog.PrologIcons;
import com.intellij.openapi.module.ModuleType;
import com.intellij.openapi.module.ModuleTypeManager;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class PrologModuleType extends ModuleType<PrologModuleBuilder> {

    private static final String ID = "PROLOG_MODULE_TYPE";

    public PrologModuleType() {
        super(ID);
    }

    public static PrologModuleType getInstance() {
        return (PrologModuleType) ModuleTypeManager.getInstance().findByID(ID);
    }

    @NotNull
    @Override
    public PrologModuleBuilder createModuleBuilder() {
        return new PrologModuleBuilder();
    }

    @NotNull
    @Override
    public String getName() {
        return "Prolog";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "GNU Prolog Module Type";
    }

    @Override
    public Icon getNodeIcon(boolean isOpened) {
        return PrologIcons.FILE;
    }
}

File: src/ch/eif/intelliprolog/module/PrologModuleType.kt
package ch.eif.intelliprolog.module

import ch.eif.intelliprolog.PrologFileType
import ch.eif.intelliprolog.PrologIcons
import com.intellij.openapi.module.ModuleType
import javax.swing.Icon

class PrologModuleType : ModuleType<PrologModuleBuilder>("PROLOG_MODULE") {

    override fun createModuleBuilder(): PrologModuleBuilder {
        return PrologModuleBuilder()
    }

    override fun getName(): String {
        return "Prolog Module"
    }

    override fun getDescription(): String {
        return "Prolog Module"
    }

    override fun getIcon(): Icon {
        return PrologIcons.FILE
    }

    override fun getNodeIcon(isOpened: Boolean): Icon {
        return PrologIcons.FILE
    }

    companion object {
        val INSTANCE: PrologModuleType = PrologModuleType()
    }

}
